### DISCUSSION TITLE
Plasticity models - General questions

### DISCUSSION THREAD

**Discussion by Nuriacor** [idaholab/moose#19232] [Q&A Modules: Solid mechanics] [answered]
Dear MOOSE developers,
I am relatively new in MOOSE and to modelling in general. I am sorry if this is too simple.
I was looking to the inelastic documentation of TensorMechanics (https://mooseframework.inl.gov/source/materials/ComputeMultipleInelasticStress.html) which was really helpful, but I have some questions.
1. Is there a difference between plastic and inelastic? Some objects are called `PlasticBlabla.C` and others `InelasticBlabla.C`
2. Looks like in the MOOSE script (.i file) there are different way to apply plasticity: Using `UserObjects` and `Materials` or just using `Materials`. Why is that? Also what `TensorMechanicsHardeningConstant` function? It looks like is use to define the constant parameters of the inelastic laws but why these values cannot be assigned directly?
3. When Multiple inelastic models are set, this means that different inelastic laws are applied to different blocks of the mesh? 
Thanks and Im sorry if this is too simple!
Nuria

**Comment by recuero** [comment, marked as answer]
1. Inelastic can be creep deformation, plastic deformation, or both. So "inelastic" is a general term for non-elastic behavior.
2. Depends on the code design. Most of the constitutive equations use `Materials`. I believe crystal plasticity may use `UserObjects` too. User objects have a ton of applications and can serve many different purposes.
3. Multiple inelastic models take care of multiple inelastic models within the same block. For example, you could think of a simulation with creep and plasticity. One needs to use a proper return mapping algorithm and, in addition, make the creep solution and the plasticity solution converge.

**Reply by Nuriacor** [reply]
Thanks!! This is really helpful!
